compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/TilingInterfaceImpl.cpp

```diff
@@ -1288,26 +1289,35 @@ Im2colOp::getTiledImplementation(OpBuilder &builder,
                        outputSlice->result_type_end());
   }
 
+  // Adjust m_offset and k_offset by adding the offsets from tiling.
+  SmallVector<OpFoldResult> newKOffsets, newMOffsets;
   AffineExpr d0, d1;
   bindDims(getContext(), d0, d1);
   auto map = AffineMap::get(2, 0, {d0 + d1}, getContext());
-  OpFoldResult kTileOffset = offsets.back();
-  OpFoldResult kOpOffset = getMixedKOffset()[0];
-  OpFoldResult kOffset = affine::makeComposedFoldedAffineApply(
-      builder, loc, map, {kTileOffset, kOpOffset});
-  OpFoldResult mTileOffset = offsets[offsets.size() - 2];
-  OpFoldResult mOpOffset = getMixedMOffset()[0];
-  OpFoldResult mOffset = affine::makeComposedFoldedAffineApply(
-      builder, loc, map, {mTileOffset, mOpOffset});
+  for (auto [outDim, kOffset] :
+       llvm::zip_equal(getKOutputDims(), getMixedKOffset())) {
+    OpFoldResult kTileOffset = offsets[outDim];
+    newKOffsets.push_back(affine::makeComposedFoldedAffineApply(
+        builder, loc, map, {kTileOffset, kOffset}));
```

Comment:
IMO we can move the `addOfr` and `mulOfr` from `AggregatedOpInterfaceImpl.cpp` to `LinalgExt/Util`. Then we can reuse the methods here. We can fix it in a follow-up.